naming/dns_resolver_test.go

```diff
@@ -0,0 +1,304 @@
+	},
+	{
+		"srv.ipv4.single.fake",
+		[]*Update{newUpdateWithMD(Add, "1.2.3.4:1234", "ipv4.single.fake")},
+	},
+	{
+		"srv.ipv4.multi.fake",
+		[]*Update{
+			newUpdateWithMD(Add, "1.2.3.4:1234", "ipv4.multi.fake"),
+			newUpdateWithMD(Add, "5.6.7.8:1234", "ipv4.multi.fake"),
+			newUpdateWithMD(Add, "9.10.11.12:1234", "ipv4.multi.fake")},
+	},
+	{
+		"srv.ipv6.single.fake",
+		[]*Update{newUpdateWithMD(Add, "[2607:f8b0:400a:801::1001]:1234", "ipv6.single.fake")},
+	},
+	{
+		"srv.ipv6.multi.fake",
+		[]*Update{
+			newUpdateWithMD(Add, "[2607:f8b0:400a:801::1001]:1234", "ipv6.multi.fake"),
+			newUpdateWithMD(Add, "[2607:f8b0:400a:801::1002]:1234", "ipv6.multi.fake"),
+			newUpdateWithMD(Add, "[2607:f8b0:400a:801::1003]:1234", "ipv6.multi.fake"),
+		},
+	},
+}
+
+var (
+	targetTc = realTargetTc
+)
+
+var hostLookupTbl = map[string][]string{
+	"foo.bar.com":      {"1.2.3.4", "5.6.7.8"},
+	"ipv4.single.fake": {"1.2.3.4"},
+	"ipv4.multi.fake":  {"1.2.3.4", "5.6.7.8", "9.10.11.12"},
+	"ipv6.single.fake": {"2607:f8b0:400a:801::1001"},
+	"ipv6.multi.fake":  {"2607:f8b0:400a:801::1001", "2607:f8b0:400a:801::1002", "2607:f8b0:400a:801::1003"},
+}
+
+var srvLookupTbl = map[string][]*net.SRV{
+	"_grpclb._tcp.srv.ipv4.single.fake": {newSRVRR("ipv4.single.fake", 1234)},
+	"_grpclb._tcp.srv.ipv4.multi.fake":  {newSRVRR("ipv4.multi.fake", 1234)},
+	"_grpclb._tcp.srv.ipv6.single.fake": {newSRVRR("ipv6.single.fake", 1234)},
+	"_grpclb._tcp.srv.ipv6.multi.fake":  {newSRVRR("ipv6.multi.fake", 1234)},
+}
+
+func updatesToSlice(updates []*Update) []Update {
+	res := make([]Update, len(updates))
+	for i, u := range updates {
+		res[i] = *u
+	}
+	return res
+}
+
+func testResolver(t *testing.T, freq time.Duration, slp time.Duration) {
+	for _, a := range targetTc {
+		r, err := NewDNSResolverWithFreq(freq)
+		if err != nil {
+			t.Fatalf("%v\n", err)
+		}
+		w, err := r.Resolve(a.target)
+		if err != nil {
+			t.Fatalf("%v\n", err)
+		}
+		var updates []*Update
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				u, err := w.Next()
+				if err != nil {
+					return
+				}
+				updates = u
+			}
+		}()
+		// Sleep for sometime to let watcher do more than one lookup
+		time.Sleep(slp)
+		w.Close()
+		wg.Wait()
+		if !reflect.DeepEqual(converToMap(a.want), converToMap(updates)) {
+			t.Errorf("Resolve(%s) = %+v, want %+v\n", a.target, updatesToSlice(updates), updatesToSlice(a.want))
+		}
+	}
+}
+
+func TestResolve(t *testing.T) {
+	// Test with real lookup functions (i.e. net.LookupHost, net.LookupSRV) and real addresses.
+	//TODO(yuxuanli): Do we need to test with net.LookupHost, net.LookupSRV and real target? If not, delete this.
+	testResolver(t, time.Millisecond*500, time.Second*1)
+
+	// Test with mocked address lookup functions and made-up addresses.
+	rp := replaceNetFunc()
+	defer rp()
+	testResolver(t, time.Millisecond*5, time.Millisecond*10)
+}
+
+const colonDefaultPort = ":" + defaultPort
+
+var IPAddrs = []struct {
```

Comment:
This should go inside the Test function (and not be named with caps at the start).